### PR TITLE
refactor(VForm): convert to TypeScript

### PR DESCRIPTION
**Pull request by sh7dm**
## Description
Convert `VForm` to TypeScript

## Motivation and Context
Convert framework to TypeScript

## How Has This Been Tested?
`jest`

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VForm/VForm.ts [hunk 1]
```diff
@@ -121,8 +124,8 @@ export default {
         novalidate: true
       }, this.$attrs),
       on: {
-        submit: e => this.$emit('submit', e)
+        submit: (e: Event) => this.$emit('submit', e)
       }
     }, this.$slots.default)
   }
```
Comment (DRoet) [inline packages/vuetify/src/components/VForm/VForm.ts]: `render` is missing the `VNode` return type?

File: packages/vuetify/src/components/VForm/VForm.ts [hunk 2]
```diff
@@ -94,19 +97,19 @@ export default {
         }, 0)
```
Comment (DRoet) [inline packages/vuetify/src/components/VForm/VForm.ts]: return types might be useful for the public methods? so `bool` for `validate()` and `void` for the others?

File: packages/vuetify/src/components/VForm/VForm.ts [hunk 3]
```diff
@@ -94,19 +97,19 @@ export default {
...
+    unregister (input: any) {
       const found = this.inputs.find(i => i._uid === input._uid)
 
       if (!found) return
 
       const unwatch = this.watchers.find(i => i._uid === found._uid)
-      unwatch.valid && unwatch.valid()
-      unwatch.shouldValidate && unwatch.shouldValidate()
+      unwatch && unwatch.valid && unwatch.valid()
+      unwatch && unwatch.shouldValidate && unwatch.shouldValidate()
```
Comment (DRoet) [inline packages/vuetify/src/components/VForm/VForm.ts]: isn't `unwatch` always set tho?

Comment (sh7dm) [reply packages/vuetify/src/components/VForm/VForm.ts]: @DRoet I did it to make TS happy

**Comment by sh7dm** [comment]
@DRoet Thanks for your feedback!

File: packages/vuetify/src/components/VForm/VForm.ts [hunk 4]
```diff
@@ -94,35 +100,35 @@ export default {
...
+      unwatch!.shouldValidate && unwatch!.shouldValidate()
 
       this.watchers = this.watchers.filter(i => i._uid !== found._uid)
       this.inputs = this.inputs.filter(i => i._uid !== found._uid)
       this.$delete(this.errorBag, found._uid)
     }
   },
 
-  render (h) {
+  render (h: CreateElement): VNode {
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VForm/VForm.ts]: remove `: CreateElement`

**Pull request merged**